src/applications/hca/containers/IDPage.jsx

```diff
@@ -6,26 +6,56 @@ import FormTitle from 'platform/forms-system/src/js/components/FormTitle';
 import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
+import { setData } from 'platform/forms-system/src/js/actions';
+import { getNextPagePath } from 'platform/forms-system/src/js/routing';
 import { focusElement } from 'platform/utilities/ui';
 import { toggleLoginModal } from 'platform/site-wide/user-nav/actions';
-import { isProfileLoading } from 'platform/user/selectors';
+import { isLoggedIn, isProfileLoading } from 'platform/user/selectors';
 
 import IDForm from '../components/IDForm';
 
 import { submitIDForm } from '../actions';
 
 class IDPage extends React.Component {
   componentDidMount() {
+    // Redirect to intro if a logged in user navigated to this page
+    // from another form page.
+    if (this.props.shouldRedirect) this.props.router.push('/');
     focusElement('.va-nav-breadcrumbs-list');
   }
 
   componentDidUpdate() {
-    // there's no need for logged-in users to see this page
-    if (this.props.shouldHideIDForm) {
-      this.props.router.push('/');
+    const { enrollmentStatus, form, hasOptionalDD214Upload } = this.props;
+
+    // Redirect to intro if a logged in user directly accessed this page.
+    if (this.props.shouldRedirect) this.props.router.push('/');
+
+    const shouldSetDD214UploadFlag =
+      hasOptionalDD214Upload && !('view:hasOptionalDD214Upload' in form.data);
+
+    if (shouldSetDD214UploadFlag) {
+      this.props.setData({ ...form.data, 'view:hasOptionalDD214Upload': true });
```

Comment:
This is pretty hacky, but I figured using the `depends` callback for the DD214 upload page would be the most conventional way to determine whether we need to show it. I injected this flag in order for that mechanism to work, should you decide to go that route when implementing that form page.

Again, I feel that perhaps ideally, the form pages should be able to look at other slices of state for conditional rendering, but that could've been a potentially messy change.